src/Catty/Parser&Serialializer/ProjectMigrator.swift

```diff
@@ -0,0 +1,114 @@
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class ProjectMigrator: ProjectMigratorProtocol {
+
+    private let correctVersionNumber: Float = 0.992
+    private var projectPath: String = ""
+    var fileManager: CBFileManager!
+
+    enum ProjectResouce: String {
+        case automaticScreenShot = "automatic_screenshot.png"
+        case maualScreenShot = "manual_screenshot.png"
+        case screenShot = "screenshot.png"
+        case imagesDirectory = "images"
+        case soundDirectory = "sounds"
+    }
+
+    init(fileManager: CBFileManager) {
+        self.fileManager = fileManager
+    }
+
+    func migrateToScene(project: Project) throws {
+
+        guard let versionNumber = Float(project.header.catrobatLanguageVersion) else {
+            throw ProjectMigratorError.unknown(description: "Unable to convert version number to Float")
+        }
+
+        if versionNumber < self.correctVersionNumber {
+            throw ProjectMigratorError.invalidCatrobatLanguageVersion
+        }
+
+        self.projectPath = project.projectPath()
+
+        let directoryPath = "\(projectPath + kLocalizedScene) 1"
+
+        if !fileManager.directoryExists(directoryPath) {
+            fileManager.createDirectory(directoryPath)
+        }
+
+        if fileManager.directoryExists(directoryPath) {
+
+            do {
+                try self.migrateToSceneDirectory(for: .automaticScreenShot, isFile: true, shouldThrowIfRequired: true)
+                try self.migrateToSceneDirectory(for: .imagesDirectory, isFile: false, shouldThrowIfRequired: true)
+                try self.migrateToSceneDirectory(for: .soundDirectory, isFile: false, shouldThrowIfRequired: true)
+                try self.migrateToSceneDirectory(for: .maualScreenShot, isFile: true, shouldThrowIfRequired: false)
+                try self.migrateToSceneDirectory(for: .screenShot, isFile: true, shouldThrowIfRequired: false)
+            } catch {
+                throw error
+            }
+
+        } else {
+            throw ProjectMigratorError.unknown(description: "Unable to create \(kLocalizedScene) directory")
+        }
+
+    }
+
+    private func migrateToSceneDirectory(for projectResouce: ProjectResouce, isFile: Bool, shouldThrowIfRequired: Bool) throws {
```

Comment:
I am using this `isFile` argument to check whether its a directory or a file that is to be moved to a new `/Scene` directory since it is required at the time of moving the file/directory which uses `moveExistingFile` and ` moveExistingDirectory` function of the `CBFileManager` respectively.

Please correct me if there is a better alternative.